homeassistant/components/overkiz/alarm_control_panel.py

```diff
@@ -0,0 +1,306 @@
+) -> None:
+    """Set up the Overkiz alarm control panel from a config entry."""
+    data: HomeAssistantOverkizData = hass.data[DOMAIN][entry.entry_id]
+    entities: list[OverkizAlarmControlPanel] = []
+
+    for device in data.platforms[Platform.ALARM_CONTROL_PANEL]:
+        if description := SUPPORTED_DEVICES.get(device.widget) or SUPPORTED_DEVICES.get(
+            device.ui_class
+        ):
+            entities.append(
+                OverkizAlarmControlPanel(
+                    device.device_url,
+                    data.coordinator,
+                    description,
+                )
+            )
+
+    async_add_entities(entities)
+
+
+class OverkizAlarmControlPanel(OverkizDescriptiveEntity, AlarmControlPanelEntity):
+    """Representation of an Overkiz Alarm Control Panel."""
+
+    entity_description: OverkizAlarmDescription
+
+    def __init__(
+        self,
+        device_url: str,
+        coordinator: OverkizDataUpdateCoordinator,
+        description: EntityDescription,
+    ) -> None:
+        """Initialize the device."""
+        super().__init__(device_url, coordinator, description)
+
+        self._attr_supported_features = self.entity_description.supported_features
+
+    @property
+    def state(self) -> str:
+        """Return the state of the device."""
+        return self.entity_description.fn_state(self.executor.select_state)
+
+    async def async_alarm_disarm(self, code: str | None = None) -> None:
+        """Send disarm command."""
+        await self.async_execute_command(
+            self.entity_description.alarm_disarm,
+            self.entity_description.alarm_disarm_args,
+        )
+
+    async def async_alarm_arm_home(self, code: str | None = None) -> None:
+        """Send arm home command."""
+        await self.async_execute_command(
+            self.entity_description.alarm_arm_home,
+            self.entity_description.alarm_arm_home_args,
+        )
+
+    async def async_alarm_arm_night(self, code: str | None = None) -> None:
+        """Send arm night command."""
+        await self.async_execute_command(
+            self.entity_description.alarm_arm_night,
+            self.entity_description.alarm_arm_night_args,
+        )
+
+    async def async_alarm_arm_away(self, code: str | None = None) -> None:
+        """Send arm away command."""
+        await self.async_execute_command(
+            self.entity_description.alarm_arm_away,
+            self.entity_description.alarm_arm_away_args,
+        )
+
+    async def async_alarm_trigger(self, code: str | None = None) -> None:
+        """Send alarm trigger command."""
+        await self.async_execute_command(
+            self.entity_description.alarm_trigger,
+            self.entity_description.alarm_trigger_args,
+        )
+
+    async def async_execute_command(self, command_name: str | None, *args: Any) -> None:
+        """Check command name and send to the executor."""
+        assert command_name
```

Comment:
This method should always expect a `command_name`. The assert should be placed before calling this method.
```suggestion
    async def async_execute_command(self, command_name: str, *args: Any) -> None:
        """Check command name and send to the executor."""
```

Reply:
I added the `async_execute_command` as a separate function to the OverkizAlarmControlPanel class, so that I don't have to do the assert at every function. This function will check and call `await self.executor.async_execute_command(command_name, *args)`, where is expects a command_name.

Should I remove this 'wrapper' function and just add the assert to every function?

Reply:
yeah remove the wrapper.

Reply:
way more readable indeed, thanks!